dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -145,22 +147,26 @@
     // Dependencies
     // -------------------------------------------------------------------------
 
-    private final OrganisationUnitStore organisationUnitStore;
+    private final TrackedEntityInstanceStoreServicesSupplier instanceStoreServicesSupplier;
 
-    private final StatementBuilder statementBuilder;
+    private StatementBuilder statementBuilder;
 
     public HibernateTrackedEntityInstanceStore( SessionFactory sessionFactory, JdbcTemplate jdbcTemplate,
         ApplicationEventPublisher publisher, CurrentUserService currentUserService,
-        AclService aclService, OrganisationUnitStore organisationUnitStore, StatementBuilder statementBuilder )
+        AclService aclService, TrackedEntityInstanceStoreServicesSupplier instanceStoreServicesSupplier )
     {
         super( sessionFactory, jdbcTemplate, publisher, TrackedEntityInstance.class, currentUserService, aclService,
             false );
 
-        checkNotNull( statementBuilder );
-        checkNotNull( organisationUnitStore );
+        checkNotNull( instanceStoreServicesSupplier );
 
-        this.statementBuilder = statementBuilder;
-        this.organisationUnitStore = organisationUnitStore;
+        this.instanceStoreServicesSupplier = instanceStoreServicesSupplier;
+    }
+
+    @PostConstruct
```

Comment:
`@PostConstruct` should not be used IMHO 😄 I think this leads to bad design. In order to for example test such a class in a unit test, you need to call init after having constructed an object of the class. If you forget to do that in a test your test diverges from what happens in production.

It looks to me as if HibernateTrackedEntityInstanceStore and the new TrackedEntityInstanceStoreServicesSupplier both depend on the StatementBuilder. If so let them both depend on it and Spring inject the StatementBuilder. If I did not make any mistake in reading the code we would then not need the PostConstruct 😅